dask_expr/tests/test_shuffle.py

```diff
@@ -136,3 +138,23 @@ def test_set_index(df, pdf):
 
     with pytest.raises(TypeError, match="can't be of type DataFrame"):
         df.set_index(df)
+
+
+def test_set_index_pre_sorted(pdf):
+    pdf = pdf.sort_values(by="y", ignore_index=True)
+    df = from_pandas(pdf, npartitions=10)
+    q = df.set_index("y")
+    assert_eq(q, pdf.set_index("y"))
+    check_expression_tree_is_blockwise(q.simplify().expr)
+    q = df.set_index(df.y)
+    assert_eq(q, pdf.set_index(pdf.y))
+    check_expression_tree_is_blockwise(q.simplify().expr)
```

Comment:
```suggestion
    assert all(isinstance(expr, Blockwise) for expr in q.simplify().find_operations(Expr))
```

This could probably be improved actually, see https://github.com/dask-contrib/dask-expr/pull/208